GEOMESA-1181 Cleanup command line tools scripts

aheyne opened the pull request:
Removed duplicate code, added geomesa-env configuration, added geomesa-common, added error handling to wget -O calls, added JAVA_OPTS support, changed GEOMESA_HOME warning output, updated all if statements to double brackets.

Signed-off-by: Austin Heyne <aheyne@ccri.com>

cfkelly commented on geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/accumulo/AccumuloParams.scala:
```diff
@@ -25,7 +25,7 @@ trait AccumuloRasterTableParam {
 }
 
-trait AccumuloConnectionParams extends OptionalZookeepersParam {
+trait git AccumuloConnectionParams extends OptionalZookeepersParam {
```
Remove "git".

aheyne replied:
How'd that git in there...

jahhulbert-ccri commented on docs/user/installation_and_configuration.rst:
```diff
@@ -201,6 +201,10 @@ The instructions below assume that the ``geomesa-tools-$VERSION`` directory is k
 
+.. note::
+
+    You can configure environment variables and classpath settings in geomesa-env.sh.
```
change ```geomesa-env.sh``` to ``geomesa-tools-$VERSION/bin/geomesa-env.sh``` or ```bin/geomesa-env.sh```

jahhulbert-ccri commented:
wondering if we should move ```bin/geomesa-env.sh``` to ```conf/geomesa-env.sh```  like hadoop/accumulo do. I think that's the general method so you can override the conf dir for different environments possibly.

jahhulbert-ccri commented on geomesa-tools/bin/geomesa:
```diff
@@ -121,41 +48,21 @@ elif [[ (-z "$ACCUMULO_CONF_DIR") || (-z "$HADOOP_CONF_DIR")]]; then
 GEOMESA_CP="${GEOMESA_CP}:${GEOMESA_CONF_DIR}:$(findSftDirs)"
-ACCUMULO_CP="$(findJars $ACCUMULO_LIB true)"
+# Add user defined directories to the classpath
+if [[ -z "$CUSTOM_CONF_DIR"  ]]; then
```
maybe rename this from ```CUSTOM_CONF_DIR``` to something like GEOMESA_EXTRA_CLASSPATH or something since ```CUSTOM_CONF_DIR``` I read as referring to a geomesa conf dir

jahhulbert-ccri commented on geomesa-tools/bin/geomesa:
```diff
@@ -258,5 +124,5 @@ else
     fi
     CLASSPATH="$GEOMESA_CP:$ACCUMULO_CP:$HADOOP_CP:${ZOOKEEPER_JAR}"
-    java ${GEOMESA_OPTS} -cp ${CLASSPATH} org.locationtech.geomesa.tools.accumulo.AccumuloRunner "$@"
+    java ${JAVA_CUSTOM_OPTS} ${GEOMESA_OPTS} -cp ${CLASSPATH} org.locationtech.geomesa.tools.accumulo.AccumuloRunner "$@"
```
rename to ```CUSTOM_JAVA_OPTS```

jahhulbert-ccri commented:
@aheyne can you merge in master ?

aheyne commented:
@jahhulbert-ccri 
The geomesa-env.sh has a configuration for the geomesa-tools-$VERSION/conf directory. I could move it there and remove that configuration option but then the end users would have to go into the scripts and change the location geomesa-env is loaded from. Further, how would we find geomesa-env? This
. "${0%/*}"/../conf/geomesa-env.sh 
would work but it seems not so great.

jahhulbert-ccri commented:
@aheyne fyi this error pops up...prob need to make this file if it doesn't exist:
```
$ bin/geomesa
bin/geomesa-common.sh: line 30: /tmp/geomesa-tools-1.2.7-SNAPSHOT/logs/geomesa.err: No such file or directory

```

jahhulbert-ccri commented:
@aheyne for the ```geomesa-env.sh``` can you take a look at how accumulo/hadoop do it? I'll try to peek soon

aheyne commented:
Moved bin/geomesa-env.sh to conf/geomesa-env.sh.
Changed behavior so by default existing environment variables will not be over written. Users can change this behavior to facilitate config swaping in the conf/geomesa-env.sh.
Removed the option to set GEOMESA_HOME and GEOMESA_CONF_DIR in geomesa-env.sh as this is no longer possible. These variables can be auto configured on first run or set ahead of time.
Log file and/or directory is created if it doesn't exist.

jahhulbert-ccri commented on geomesa-tools/bin/geomesa-common.sh:
```diff
@@ -0,0 +1,220 @@
+# Find geomesa-env and load config
+GEOMESA_ENV=${GEOMESA_CONF_DIR}/geomesa-env.sh
+if [[ -f "$GEOMESA_ENV" ]]; then
+    . GEOMESA_ENV
```
i'm getting this

```ahulbert@doubletrouble:/tmp/geomesa-tools-1.2.7-SNAPSHOT$ bin/geomesa
bin/geomesa-common.sh: line 173: GEOMESA_ENV: No such file or directory```

aheyne replied:
Sorry, I forgot to commit that fix...

jahhulbert-ccri commented on geomesa-tools/bin/geomesa-common.sh:
```diff
@@ -0,0 +1,220 @@
+    echo "export PATH=\${GEOMESA_HOME}/bin:\$PATH"
+}
+
+function contiansElement() {
```
spelling...change here and in usage

jahhulbert-ccri commented on geomesa-tools/bin/download-data.sh:
```diff
@@ -23,63 +22,74 @@ setGeoHome ()
 case "$type" in
-  gdelt)
-    read -p "Enter a date in the form YYYYMMDD: " DATE
+    gdelt)
```
2 spaces i think is what and  we are doing? and if you are in vim make sure to use spaces not tabs!

aheyne replied:
We're using 2 spaces in everything else but for some reason the bash scripts were at 4 when I started working on them so I just kept that. Should I switch it all over?

jahhulbert-ccri replied:
meh yeah we should try to be consistent...doesn't matter all that much at the moment looks like we mix it up in other scripts too i guess we can fix it later

jahhulbert-ccri commented:
@aheyne is this just the one ticket?

aheyne commented:
[GEOMESA-1373](https://geomesa.atlassian.net/browse/GEOMESA-1373)
[GEOMESA-1142](https://geomesa.atlassian.net/browse/GEOMESA-1142) [GEOMESA-1265](https://geomesa.atlassian.net/browse/GEOMESA-1265) (should be working, not sure how to test)
[GEOMESA-1187](https://geomesa.atlassian.net/browse/GEOMESA-1187)
